pymc/model/core.py

```diff
@@ -964,6 +964,8 @@ def add_coord(
         if name in self.coords:
             if not np.array_equal(values, self.coords[name]):
                 raise ValueError(f"Duplicate and incompatible coordinate: {name}.")
+            assert name in self._dim_lengths
```

Comment:
assert seems silly, given the `if name in self.coords` so close by. Maybe just a comment before the return?